Px.Utils/ModelBuilders/MatrixMetadataBuilder.cs

```diff
@@ -226,20 +234,26 @@ private ContentDimension BuildContentDimension(Dictionary<MetadataEntryKey, stri
         {
             string code = GetDimensionCode(entries, langs, dimensionName);
             ContentValueList values = BuildContentDimensionValues(entries, langs, dimensionName);
-            // Table level UNIT, SHOWDECIMALS and DECIMALS properties are not needed after building the content dimension, so they're removed here
-            if (TryGetEntries(entries, _conf.Tokens.KeyWords.Units, langs, out Dictionary<MetadataEntryKey, string>? unitEntries))
-            {
-                foreach (MetadataEntryKey key in unitEntries.Keys) entries.Remove(key);
-            }
-            if (TryGetEntries(entries, _conf.Tokens.KeyWords.Decimals, langs, out Dictionary<MetadataEntryKey, string>? decimalEntries))
-            {
-                foreach (MetadataEntryKey key in decimalEntries.Keys) entries.Remove(key);
-            }
-            if (TryGetEntries(entries, _conf.Tokens.KeyWords.ShowDecimals, langs, out Dictionary<MetadataEntryKey, string>? showDecimalEntries))
+            // Table level UNIT, SHOWDECIMALS, PRECISION and DECIMALS properties are not needed after building the content dimension, so they're removed here
+            string[] keywords = [
+                _conf.Tokens.KeyWords.Units,
+                _conf.Tokens.KeyWords.ShowDecimals,
```

Comment:
Leave ShowDecimals since it is/should be presentation level information and might be useful independent on the precision in the content variable values.